[WebAuthn] Reject too long or too short user id

pascoej opened the pull request:
#### 8f49569a327273a80904ab7d3cf11b31f5fcb4cc
<pre>
[WebAuthn] Reject too long or too short user id
<a href="https://bugs.webkit.org/show_bug.cgi?id=242073">https://bugs.webkit.org/show_bug.cgi?id=242073</a>
rdar://90281685

Reviewed by Brent Fulgham.

The creation option&apos;s user.id should be between 1-64 bytes inclusive
per the spec: <a href="https://w3c.github.io/webauthn/#dom-publickeycredentialuserentity-id">https://w3c.github.io/webauthn/#dom-publickeycredentialuserentity-id</a>

* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-hid.https-expected.txt:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-hid.https.html:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-with-invalid-parameters.https-expected.txt:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-with-invalid-parameters.https.html:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp:
(WebCore::AuthenticatorCoordinator::create const):

Canonical link: <a href="https://commits.webkit.org/251938@main">https://commits.webkit.org/251938@main</a>
</pre>

brentfulgham commented on LayoutTests/http/wpt/webauthn/public-key-credential-create-with-invalid-parameters.https.html:
```diff
@@ -43,6 +43,8 @@
         // wrong user attribute type
         [rp, { name: Symbol(), id: user.id, displayName: user.displayName}, challenge, [pubKeyCredParam]],
+        [rp, { name: user.name, id: asciiToUint8Array(""), displayName: user.displayName}, challenge, [pubKeyCredParam]],
+        [rp, { name: user.name, id: asciiToUint8Array("12345678123456781234567812345678123456781234567812345678123456781"), displayName: user.displayName}, challenge, [pubKeyCredParam]],
```
It seems like we have a test case for a single-character id. Do we also have one for a 64-byte ID?

Clearly you also test the < 1 and > 64 cases in this change.

pascoej replied:
Good idea, added test cases.

pascoej committed:
[WebAuthn] Reject too long or too short user id
https://bugs.webkit.org/show_bug.cgi?id=242073
rdar://90281685

Reviewed by Brent Fulgham.

The creation option's user.id should be between 1-64 bytes inclusive
per the spec: https://w3c.github.io/webauthn/#dom-publickeycredentialuserentity-id

* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-hid.https-expected.txt:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-success-hid.https.html:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-with-invalid-parameters.https-expected.txt:
* LayoutTests/http/wpt/webauthn/public-key-credential-create-with-invalid-parameters.https.html:
* Source/WebCore/Modules/webauthn/AuthenticatorCoordinator.cpp:
(WebCore::AuthenticatorCoordinator::create const):

Canonical link: https://commits.webkit.org/251938@main
